registry/storage/manifeststore.go

```diff
@@ -124,6 +126,16 @@ func (ms *manifestStore) verifyManifest(ctx context.Context, mnfst *manifest.Sig
 	}
 
 	if !ms.skipDependencyVerification {
+		if len(mnfst.FSLayers) == 0 || len(mnfst.History) == 0 {
+			errs = append(errs, distribution.ErrManifestValidation{
+				Reason: "no layers present"})
+		}
+
+		if len(mnfst.FSLayers) != len(mnfst.History) {
```

Comment:
So based on the error, while the easily reproducable problem was tagging the same thing twice and pushing, resulting in an empty second manifest. As far as I understood the problem, if the second manifest being pushed share the n'th layer with the first pushed manifest, we would have n-1 layers present in the manifest.

I think this check should solve for that case, just double checking.